daemon/algod/server.go

```diff
@@ -84,6 +88,34 @@ func (s *Server) Initialize(cfg config.Local, phonebookAddresses []string, genes
 	s.log.SetLevel(logging.Level(cfg.BaseLoggerDebugLevel))
 	setupDeadlockLogger()
 
+	// Check some config parameters.
+	if cfg.RestConnectionsSoftLimit > cfg.RestConnectionsHardLimit {
+		s.log.Warnf(
+			"RestConnectionsSoftLimit %d exceeds RestConnectionsHardLimit %d",
+			cfg.RestConnectionsSoftLimit, cfg.RestConnectionsHardLimit)
+		cfg.RestConnectionsSoftLimit = cfg.RestConnectionsHardLimit
+	}
+	if cfg.IncomingConnectionsLimit < 0 {
+		return fmt.Errorf(
```

Comment:
this will break algod startup for people who have `IncomingConnectionsLimit=-2` for whatever reason. This value looks allowed before the proposed change.

Reply:
I think that it's reasonable to "break" a configuration that should not have existed to start with.